usr/share/rear/layout/save/GNU/Linux/200_partition_layout.sh

```diff
@@ -26,7 +26,23 @@ extract_partitions() {
     declare sysfs_name=$(get_sysfs_name $device)
 
     ### check if we can find any partitions
-    declare -a sysfs_paths=(/sys/block/$sysfs_name/$sysfs_name*)
+    declare -a sysfs_paths_unfiltered=(/sys/block/$sysfs_name/$sysfs_name*)
+
+    ### initialize array
+    declare -a sysfs_paths=()
+
+    ### Silently skip invalid partitions on eMMC devices
+    ### of type *rpmb or *boot[0-9] like /dev/mmcblk0rpmb or /dev/mmcblk0boot0
+    ### because sysfs for (some?) eMMC disks recognises partitions,
+    ### which are no usual partitions, but special areas on the eMMC.
+    ### When trying to get the partition layout for such disks, ReaR would exit with an error.
+    ### Therefore we ignore these special partitions, see also
+    ### https://github.com/rear/rear/issues/2087
+    for possible_sysfs_partition in ${sysfs_paths_unfiltered[@]}; do
```

Comment:
Usually one should use `"${array[@]}"` with double-quotes
because using `${array[@]}` without double-quotes is problematic,
see 'Arrays' in "man bash" and see
https://github.com/rear/rear/issues/1068
for some examples.

Reply:
Fixed.